services/galley/src/Galley/Cassandra/Conversation.hs

```diff
@@ -52,17 +52,20 @@ import qualified UnliftIO
 import Wire.API.Conversation hiding (Conversation, Member)
 import Wire.API.Conversation.Role (roleNameWireAdmin)
 
-createConversation :: NewConversation -> Client Conversation
-createConversation (NewConversation ty usr acc arole name mtid mtimer recpt users role) = do
+createConversation :: Local x -> NewConversation -> Client Conversation
+createConversation loc (NewConversation ty usr acc arole name mtid mtimer recpt users role protocol) = do
   conv <- Id <$> liftIO nextRandom
-  retry x5 $ case mtid of
-    Nothing ->
-      write Cql.insertConv (params LocalQuorum (conv, ty, usr, Cql.Set (toList acc), Cql.Set (toList arole), fmap fromRange name, Nothing, mtimer, recpt))
-    Just tid -> batch $ do
-      setType BatchLogged
-      setConsistency LocalQuorum
-      addPrepQuery Cql.insertConv (conv, ty, usr, Cql.Set (toList acc), Cql.Set (toList arole), fmap fromRange name, Just tid, mtimer, recpt)
-      addPrepQuery Cql.insertTeamConv (tid, conv)
+  let groupId = case protocol of
+        ProtocolProteus -> Nothing
+        ProtocolMLS -> Just . convIdToGroupId . qUntagged . qualifyAs loc $ conv
+  retry x5 . batch $ do
+    setType BatchLogged
+    setConsistency LocalQuorum
+    addPrepQuery
+      Cql.insertConv
+      (conv, ty, usr, Cql.Set (toList acc), Cql.Set (toList arole), fmap fromRange name, mtid, mtimer, recpt, protocol, groupId)
+    for_ mtid $ \tid -> addPrepQuery Cql.insertTeamConv (tid, conv)
+    for_ groupId $ \gid -> addPrepQuery Cql.insertGroupId (gid, conv, tDomain loc)
```

Comment:
What's the point of `decodeConvId` if we're storing the group ID mapping for local conversations?

Reply:
I dropped `decodeConvId` compared to the last time you review this PR.